deps: Move canvas to development dependencies

kyranet opened the pull request:
More information about devDependencies [here](https://docs.npmjs.com/files/package.json#devdependencies).

The reason for this change is that gifencoder does not seem to need the package `canvas` to work as shown in the [addFrame lines](https://github.com/eugeneware/gifencoder/blob/master/lib/GIFEncoder.js#L189-L190), but rather checks for the existence of the method. However, I do not find any `require('canvas')` in the project (unless I'm missing something!) besides the example files and tests.

kyranet committed:
deps: Move canvas to development dependencies

relsqui commented:
I'd love it if this got merged, it's currently blocking me upgrading to 2.0.0 because I don't want to debug canvas's install failures in CI given that I don't actually need canvas. :)

cenfun commented:
this is great, since "canvas" module was so big (65.74 MB) and it's not runtime dependency

lostfictions commented:
would love for this to get merged!

kyranet commented:
I really didn't want to come down to this, as I'm very against promoting my own code in similar packages, but since there's virtually no update in this repository, and this could help a lot of people, I have made a fork and is available at [`@skyra/gifenc`](https://www.npmjs.com/package/@skyra/gifenc).

I have rewritten the entire package to TypeScript (types are shipped in the package, no need for `@types/`), removed several global variables, improved performance across the board (2x speedup), all while keeping nearly 1:1 functionality and behaviour, it should be a drop-in replacement.
